compiler/AST/foralls.cpp

```diff
@@ -594,55 +632,53 @@ static bool findStandaloneOrLeader(ForallStmt* pfs, CallExpr* iterCall)
 }
 
 static void addParIdxVarsAndRestruct(ForallStmt* fs, bool gotSA) {
-  // First, save away the user loop as its own BlockStmt.
+  if (gotSA) {
+    // No need to restructure anything. Leaving it as-is for simplicity.
+
+    VarSymbol* parIdx = parIdxVar(fs);
+
+    // FLAG_INDEX_OF_INTEREST is needed in setConstFlagsAndCheckUponMove():
+    parIdx->addFlag(FLAG_INDEX_OF_INTEREST);
+    parIdx->addFlag(FLAG_INDEX_VAR);
+
+    return;
+  }
+
+  // Keep the user loop as its own BlockStmt.
   // Make it the last thing in the new fs->loopBody().
   BlockStmt* userLoopBody = fs->loopBody();
   BlockStmt* newLoopBody = new BlockStmt();
   userLoopBody->replace(newLoopBody);
   newLoopBody->insertAtTail(userLoopBody);
 
-  // Now, add parIdx*.
-  //
-  // These are the original buildForallLoopStmt(), with "lead" -> "par".
-  // We could retain the original names i.e. gotSA ? saIdx : leadIdx.
-  VarSymbol* parIdx     = newTemp("chpl__parIdx");
-  VarSymbol* parIdxCopy = newTemp("chpl__parIdxCopy");
+  // The induction variable of the parallel loop.
+  VarSymbol* parIdx = newTemp("chpl_followThis");
 
-  // This is how these flags have been set historically.
-  // Todo make them the same regardless of gotSA.
-  if (gotSA) {
-    // This is needed in setConstFlagsAndCheckUponMove():
-    parIdx->addFlag(FLAG_INDEX_OF_INTEREST);
+  // If there is only one follower, we are tempted to use
+  // the original forall's induction variable as the
+  // the induction variable of the follower loop.
+  // Alas, this results in the autoDestroy for that variable
+  // to be inserted outside the loop, and more trouble from that.
+  // Ex. test/functions/ferguson/ref-pair/iterating-over-arrays.chpl
```

Comment:
I'd expect leader/follower iteration to be represented directly in the ForallStmt rather than partially lowered here. I can certainly see this PR as an intermediate step but am wondering if I'm understanding the situation correctly and if you agree with the goal.